src/services/completions.ts

```diff
@@ -1519,7 +1520,20 @@ namespace ts.Completions {
                         return false;
                     }
 
-                    symbol = skipAlias(symbol, typeChecker);
+                    // External modules can have global export declarations that will be
+                    // available as global keywords in all scopes. But if the external module
+                    // already has an explicit export and user only wants to user explicit

```

Comment:
I don’t see you checking for whether the file is a module (`sourceFile.externalModuleIndicator`), so it looks like this will apply to scripts as well, which I think we don’t want. You should add another test for that (completions in a file with no imports or exports).

Reply:
It seems if the file is a script we never offer auto import at all by running `shouldOfferImportCompletions()`. I'll add the test.

Reply:
What other case should we cover? I tried something the example below and there was no Auto Import provided
```ts
const el = class/*1*/
```

Reply:
> It seems if the file is a script we never offer auto import at all by running `shouldOfferImportCompletions()`

You’ll want to add some other way of making that function return true, like setting `module` to `es2015`.

Reply:
Ya thanks for pointing that out. It should be ok now with the last commit. Please review that